core-test/src/test/scala/org/bitcoins/core/wallet/utxo/UTXOSpendingInfoTest.scala

```diff
@@ -0,0 +1,204 @@
+package org.bitcoins.core.wallet.utxo
+
+import org.bitcoins.core.crypto.ECPrivateKey
+import org.bitcoins.core.currency.CurrencyUnits
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script._
+import org.bitcoins.core.protocol.transaction.{
+  BaseTransaction,
+  TransactionConstants,
+  TransactionOutPoint,
+  TransactionOutput
+}
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.testkit.core.gen.ScriptGenerators
+import org.scalatest.AsyncFlatSpec
+
+class UTXOSpendingInfoTest extends AsyncFlatSpec {
+
+  def randomSPK: ScriptPubKey = {
+    ScriptGenerators.scriptPubKey.map(_._1).sample.get
+  }
+
+  def randomWitnessSPK: WitnessScriptPubKeyV0 = {
+    ScriptGenerators.witnessScriptPubKeyV0.map(_._1).sample.get
+  }
+
+  behavior of "UTXOSpendingInfo"
+
+  it must "fail given a bad P2SH script" in {
+    val privKey = ECPrivateKey.freshPrivateKey
+    val pubKey = privKey.publicKey
+    val p2sh = P2SHScriptPubKey(P2PKScriptPubKey(pubKey))
+    val creditingOutput = TransactionOutput(CurrencyUnits.zero, p2sh)
+    val creditingTx = BaseTransaction(version =
```

Comment:
nit: I think you can use `TransactionGenerators.buildCreditingTransaction()` to remove some of this boiler plate.